src/main/java/com/oath/halodb/HaloDBInternal.java

```diff
@@ -281,6 +285,91 @@ int get(byte[] key, ByteBuffer buffer) throws IOException {
         }
     }
 
+    public boolean takeSnapShot() {
+        synchronized (snapshotLock) {
```

Comment:
```synchronized``` method can achieve same impact but clear context if we don't have other methods defined as ```synchronized```

Reply:
There's another method in using synchronized in class level. It's better to use a separate lock for this function to avoid potential blocking of other methods.

Reply:
Consider put ```synchronized``` to ```open()```, ```close``` and ```snapshot``` to prevent from concurrent operations. And you can use a separate lock for ```resetStats``` @amannaly Any suggestion?